feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,753 @@
+	cardList := append(controllerCards, lineCards...)
+	for _, component := range cardList {
+		query := gnmi.OC().Component(component).State()
+		timestamp := time.Now().Round(time.Second)
+		componentState := gnmi.Get(t, dut, query)
+		componentType := componentState.GetType()
+		if componentType == lineCardType && deviations.LinecardMemoryUtilizationUnsupported(dut) {
+			t.Logf("INFO: Skipping test for linecard component %s due to deviation linecard_memory_utilization_unsupported", component)
+			continue
+		}
+
+		memoryState := componentState.GetMemory()
+		if memoryState == nil {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Memory data not available",
+				timestamp, deviceName, description, component, componentType)
+			continue
+		}
+
+		memoryAvailable := memoryState.GetAvailable()
+		memoryUtilized := memoryState.GetUtilized()
+		memoryUtilization := uint8((memoryUtilized * 100) / (memoryAvailable + memoryUtilized))
+		t.Logf("INFO: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%%", timestamp, deviceName, description, component, componentType, memoryUtilization)
+	}
+}
+
+func TestSystemProcessNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	const description = "System CPU Process"
+
+	timestamp := time.Now().Round(time.Second)
+	results := gnmi.GetAll(t, dut, query)
+	for _, result := range results {
+		processName := result.GetName()
+
+		if result.CpuUtilization == nil {
+			t.Errorf("%s %s ERROR: %s process %-40s utilization not available", timestamp, deviceName, description, processName)
+		} else {
+			t.Logf("%s %s INFO: %s process %-40s utilization: %3d%%", timestamp, deviceName, description, processName, result.GetCpuUtilization())
+		}
+	}
+}
+
+func TestSystemProcessNoHighMemorySpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	const description = "System Process Memory"
+
+	currentTime := time.Now().Round(time.Second)
+	processes := gnmi.GetAll(t, dut, query)
+	for _, process := range processes {
+		processName := process.GetName()
+		if process.MemoryUtilization != nil {
+			t.Logf("%s %s INFO:  %s - Process: %-40s - Utilization: %3d%%", currentTime, deviceName, description, processName, process.GetMemoryUtilization())
+		} else {
+			t.Errorf("%s %s ERROR:  %s - Process: %-40s - Utilization data not available", currentTime, deviceName, description, processName)
+		}
+	}
+}
+
+func TestNoQueueDrop(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	type testCase struct {
+		desc     string
+		path     string
+		counters []*ygnmi.Value[uint64]
+	}
+	interfaces := sortedInterfaces(dut.Ports())
+	t.Logf("Interfaces: %s", interfaces)
+	for _, intf := range interfaces {
+		qosInterface := gnmi.OC().Qos().Interface(intf)
+		var cases []testCase
+		switch dut.Vendor() {
+		case ondatra.JUNIPER:
+			cases = []testCase{
```

Comment:
The only difference between the three vendors, that I see is that `input/queues/queue/state/dropped-pkts` is not supported in Juniper.

Can you remove the switch case and just skip that leaf for Juniper?

Reply:
Thanks for your comments. Remove the switch case and add skip leaf for Juniper:

```go
					if dut.Vendor() == ondatra.JUNIPER && c.desc == "Queue Input Dropped packets" {
						t.Skipf("INFO: Skipping test due to %s does not support %s", dut.Vendor(), c.path)
					}
```